test/e2e-go/upgrades/rekey_support_test.go

```diff
@@ -79,35 +82,49 @@ func TestRekeyUpgrade(t *testing.T) {
 	a.NoError(err)
 
 	tx.RekeyTo = addrB
-
 	rekey, err := client.SignTransactionWithWalletAndSigner(wh, nil, "", tx)
 	a.NoError(err)
 
 	_, err = client.BroadcastTransaction(rekey)
-	a.Error(err)
-	// should be either "transaction has RekeyTo set but rekeying not yet enable" or "txn dead"
-	if !strings.Contains(err.Error(), "transaction has RekeyTo set but rekeying not yet enable") &&
-		!strings.Contains(err.Error(), "txn dead") {
-		a.NoErrorf(err, "error message should be one of :\n%s\n%s", "transaction has RekeyTo set but rekeying not yet enable", "txn dead")
+	// non empty err means the upgrade have not happened yet (as expected), ensure the error
+	if err != nil {
+		// should be either "transaction has RekeyTo set but rekeying not yet enable" or "txn dead"
+		if !strings.Contains(err.Error(), "transaction has RekeyTo set but rekeying not yet enable") &&
+			!strings.Contains(err.Error(), "txn dead") {
+			a.NoErrorf(err, "error message should be one of :\n%s\n%s", "transaction has RekeyTo set but rekeying not yet enable", "txn dead")
+		}
+	} else {
+		// if we had no error it must mean that we've upgraded already. Verify that.
+		curStatus, err := client.Status()
+		a.NoError(err)
+		a.NotEqual(consensusTestUnupgradedProtocol, protocol.ConsensusVersion(curStatus.LastVersion))
```

Comment:
If this is true, the subsequent test should be skipped, and you could skip ahead to the point where we don't need to wait for the upgrade.

Reply:
I agree but I found keeping the test logic sequential is better for readability than deep nested logic or goto.